src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -0,0 +1,390 @@
+    sage: p2 = R(f); p2
+    1*a*b + 1*a + 1*b
+
+Some basic arithmetic operations::
+
+    sage: p1 + p2
+    3*a*b + 1*a + 1*b
+    sage: p1 * p2
+    4*a^2*b^2 + 4*a^2*b + 1*a^2 + 4*a*b^2 + 1*a*b + 0*b^2
+    sage: T(2) * p1
+    5*a*b + 2*a + 1*b
+    sage: p1(T(1),T(2))
+    6
+
+Let's look at the different result for tropical curve and graph of tropical
+polynomial in two variables when the min-plus or max-plus algebra is used:
+
+    sage: T = TropicalSemiring(QQ, use_min=True)
+    sage: R = PolynomialRing(T, 'a,b')
+    sage: dict1 = {(1,0):0, (0,1):-1, (1,1):3}
+    sage: p1 = R(dict1)
+    sage: p1.tropical_hypersurface()
+    Tropical curve of 3*a*b + 0*a + (-1)*b are 
+    [[(t1 - 1, t1), [t1 >= -3], 1]
+    [(t1, -3), [t1 <= -4], 1]
+    [(-4, t1), [t1 <= -3], 1]]
+    sage: plot(p1.tropical_hypersurface())
+    sage: p1.plot3d()
+
+    sage: T = TropicalSemiring(QQ, use_min=False)
+    sage: R = PolynomialRing(T, 'a,b')
+    sage: dict1 = {(1,0):0, (0,1):-1, (1,1):3}
+    sage: p1 = R(dict1)
+    sage: p1.tropical_hypersurface()
+    Tropical curve of 3*a*b + 0*a + (-1)*b are 
+    [[(t1 - 1, t1), [t1 <= -3], 1]
+    [(t1, -3), [t1 >= -4], 1]
+    [(-4, t1), [t1 >= -3], 1]]
+    sage: plot(p1.tropical_hypersurface())
+    sage: p1.plot3d()
+
+TESTS:
+
+    sage: -p1
+    Traceback (most recent call last):
+    ...
+    ArithmeticError: cannot negate any non-infinite element
+
+REFERENCES:
+
+    - [Bru2013]_
+    - [Fil2017]_
+
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2024 Verrel Rievaldo Wijaya <verrelrievaldo@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.misc.cachefunc import cached_method
+
+from sage.structure.parent import Parent
+from sage.structure.unique_representation import UniqueRepresentation
+from sage.plot.plot3d.list_plot3d import list_plot3d
+from sage.symbolic.ring import SR
+from sage.categories.semirings import Semirings
+
+from sage.rings.polynomial.term_order import TermOrder
+from sage.rings.polynomial.multi_polynomial import MPolynomial
+from sage.rings.polynomial.multi_polynomial_element import MPolynomial_polydict
+# from sage.rings.polynomial.polydict import ETuple
+from sage.rings.semirings.tropical_semiring import TropicalSemiring
+from sage.rings.semirings.tropical_variety import TropicalCurve, TropicalVariety
+
+class TropicalMPolynomial(MPolynomial_polydict):
+    r"""
+    Generic multivariate tropical polynomial.
+
+    """
+
+    def plot3d(self):
+        """
+        Return the 3d plot of ``self``
+
+        OUTPUT: A Graphics3d Object
+
+        EXAMPLE:
```

Comment:
```suggestion
        EXAMPLES::
```
and similar. See the dev guide.